File: javascript/darts/darts.js
export const score = (x, y) => {
  const targetRadius = Math.sqrt(x * x + y * y)

  if (targetRadius > 10) {
    return 0;
  } else if (targetRadius <= 10 && targetRadius > 5) {
    return 1;
  } else if (targetRadius <= 5 && targetRadius > 1) {
    return 5;
  } else {
    return 10;
  }
};

File: elixir/darts/lib/darts.ex
defmodule Darts do
  @type position :: {number, number}

  @doc """
  Calculate the score of a single dart hitting a target
  """
  @spec score(position) :: integer
  def score({x, y}) do
    x2 = :math.pow(x, 2)
    y2 = :math.pow(y, 2)

    x2
    |> Kernel.+(y2)
    |> :math.sqrt()
    |> calculate_score()
  end

  defp calculate_score(distance) when distance > 10, do: 0
  defp calculate_score(distance) when distance <= 10 and distance > 5, do: 1
  defp calculate_score(distance) when distance <= 5 and distance > 1, do: 5
  defp calculate_score(distance) when distance <= 1 and distance >= 0, do: 10
end
